Glossary (#5546)
withastro/docs: contributor-guides/glossary.md

## Islands

Islands is a frontend architecture that results in better frontend performance by helping you avoid monolithic JavaScript patterns and stripping all non-essential JavaScript from the page automatically. The technique that this architectural pattern builds on is also known as partial or selective hydration. In Astro, an “island” refers to any interactive UI component on the page. Read more about islands in the [Astro Islands](https://docs.astro.build/en/concepts/islands/) page.

-----

SARAH RESOLVING CONFLICTS
Additions from this PR are added below the existing content

-----

> ## Adaptors
> 
> Adaptors are essential for deploying Astro projects to different hosting platforms. Each adaptor is tailored to a specific host, such as Netlify, Vercel, or Cloudflare Pages, ensuring compatibility and streamlined deployment processes. Adaptors handle the nuances of each platform, providing a seamless integration for Astro projects.

Reviewer: ```suggestion
## Adapters

Adapters are essential for deploying Astro projects to different hosting platforms. Each adapter is tailored to a specific host, such as Netlify, Vercel, or Cloudflare Pages, ensuring compatibility and streamlined deployment processes. Adapters handle the nuances of each platform, providing a seamless integration for Astro projects.
```

Just some typos :)

For more details, see:

> - [Astro Adaptor API Reference](https://docs.astro.build/en/reference/adapter-reference/)

Reviewer: ```suggestion
- [Astro Adapter API Reference](https://docs.astro.build/en/reference/adapter-reference/)
```

> ## API

Reviewer: Not a change, but just a note that we will need to classify these based on Astro-relatedness at some point in the future.

An API (Application Programming Interface) in the context of Astro is a set of tools and protocols used for building and interacting with software applications. Astro's API provides developers with functions, methods, and standards to create dynamic web applications. This includes various components, utilities, and integrations that facilitate interaction between the Astro framework and other services or applications. By leveraging Astro's API, developers can efficiently build, extend, and customize their web projects, tapping into a rich ecosystem of web development functionalities.

For more details, see:
- [Astro API Reference](https://docs.astro.build/en/reference/api-reference/)

## Build

Building refers to the process of compiling an Astro project into static files. This step transforms components, scripts, and styles into optimized HTML, CSS, and JavaScript, preparing them for deployment. The build process includes essential optimizations like minification and code splitting, crucial for enhancing performance.

For more details, see:
- [Astro Build Command](https://docs.astro.build/en/reference/cli-reference/#astro-build)

> - [Deploy your Astro Site](https://docs.astro.build/en/guides/deploy/)

Reviewer: ```suggestion
- [Guide: Deploy your Astro Site](https://docs.astro.build/en/guides/deploy/)
```

## Client Side Rendering

Client Side Rendering involves rendering web components in the browser through JavaScript. Unlike server-side rendering, client side rendering is used for dynamic content that updates in response to user interactions, providing a more interactive and responsive experience.

For more details, see:
- [Client-Side Scripts](https://docs.astro.build/en/guides/client-side-scripts/#client-side-scripts)

> - [Client Only Directive](https://docs.astro.build/en/reference/directives-reference/#clientonly)

Reviewer: ```suggestion
- [`client:only` Directive](https://docs.astro.build/en/reference/directives-reference/#clientonly)
```
To follow suit with the docs.

## Cloudflare

> Astro's integration with Cloudflare allows for deploying projects on Cloudflare's edge network, enhancing site performance and reliability. This integration benefits from Cloudflare's global distribution, offering faster content delivery to users.
> 
> For more details, see:
> - [Deploying on Cloudflare Guide](https://docs.astro.build/en/guides/deploy/cloudflare/)

Reviewer: ```suggestion
Astro's integration with Cloudflare allows for deploying projects to Cloudflare's edge network, enhancing site performance and reliability. This integration benefits from Cloudflare's global distribution, offering faster content delivery to users.

For more details, see:
- [Guide: Deploying to Cloudflare](https://docs.astro.build/en/guides/deploy/cloudflare/)
```

## Collection Entry

> A Collection Entry is a content piece within a collection directory, supporting formats like Markdown, MDX, YAML, and JSON. These entries are crucial for content organization and management projects.

Reviewer: ```suggestion
A Collection Entry is a content piece within a collection directory, supporting formats like Markdown, MDX, YAML, and JSON. 
```

This definition should probably be more in-depth. Content collections and their parts are pretty `crucial` to the Astro experience! Perhaps we can include more about how a collection entry is rendered and more about what a `collection directory` is (and perhaps a re-wording of this).

For more details, see:
- [Content Collections](https://docs.astro.build/en/guides/content-collections/)

> ## Components

Reviewer: ```suggestion
## Component
```
I will ask others their opinion, but I think the terms should be singular - like a dictionary. LMK what you think of this.

Astro components are the basic building blocks of any Astro project. They are HTML-only templating components with no client-side runtime. You can spot an Astro component by its file extension: `.astro`. 

For more details, see:
- [Components](https://docs.astro.build/en/core-concepts/astro-components/)

## Configuration

Configuration involves setting up global project settings via the `astro.config.mjs` file. This includes defining build options, customizing the development environment, and integrating with various tools and frameworks.

For more details, see:
- [Configuring Astro Guide](https://docs.astro.build/en/guides/configuring-astro/)
- [Configuration Reference](https://docs.astro.build/en/reference/configuration-reference/)

> ## Content Collections

Reviewer: ```suggestion
## Content Collection
```

> Content Collections refer to a method for organizing and managing content within an Astro project. A content collection is a top-level directory inside the `src/content` directory, like `src/content/newsletter` or `src/content/authors`. Collection entries can be in various formats, including Markdown, MDX, YAML, and JSON. These collections are integral for structuring documents, validating frontmatter, and providing TypeScript type-safety.

Reviewer: ```suggestion
Content Collections are used to organize and manage content within an Astro project. A content collection is a top-level directory inside the `src/content` directory, like `src/content/newsletter` or `src/content/authors`. Collection entries can be in various formats, including Markdown, MDX, YAML, and JSON. These collections are integral for structuring documents, validating frontmatter, and providing TypeScript type-safety.
```

For more details, see:
- [Content Collections](https://docs.astro.build/en/guides/content-collections/)

> ## CSS

Reviewer: Perhaps change this to `Scoped Styles` in general
```suggestion
## Scoped Styles
```
